Update to latest upstream (#1)
platformsh-templates/prestashop: README.md

## Features

* PHP 7.4
* MariaDB 10.4
* Composer-based build
* Automatic TLS certificates

## Customizations

The following changes have been made relative to Prestashop as it is downloaded from Prestashop.com. If using this project as a reference for your own existing project, replicate the changes below to your project.

* [.platform/routes.yaml](.platform/routes.yaml): Platform.sh allows you to define the [routes](https://docs.platform.sh/configuration/routes.html).
* [.platform/services.yaml](.platform/services.yaml):  Platform.sh allows you to completely define and configure the topology and [services you want to use on your project](https://docs.platform.sh/configuration/services.html).
* [.platform.app.yaml](.platform.app.yaml): You control your application and the way it will be built and deployed on Platform.sh [via a single configuration file](https://docs.platform.sh/configuration/app-containers.html).

## References

* [Platform.sh post](https://platform.sh/blog/2019/java-hello-world-at-platform.sh/)
* [Prestashop](https://www.prestashop.com/en)
* [PHP at Platform.sh](https://docs.platform.sh/languages/php.html)

> ## How to install

Reviewer: This section still needs to be addressed.

# Prestashop for Platform.sh

<p align="center">
<a href="https://console.platform.sh/projects/create-project?template=https://raw.githubusercontent.com/platformsh/template-builder/master/templates/prestashop/.platform.template.yaml&utm_content=prestashop&utm_source=github&utm_medium=button&utm_campaign=deploy_on_platform">
    <img src="https://platform.sh/images/deploy/lg-blue.svg" alt="Deploy on Platform.sh" width="180px" />
</a>
</p>

> This template demonstrates building a Spring Boot application. It uses Gradle to build the application and deploy it to Platform.sh.
> 

Reviewer: ```suggestion
```

> This template builds Prestashop project.  It is pre-configured to use MariaDB.

Reviewer: Anything else we can say about it here?

> PrestaShop is a freemium, open source e-commerce solution. The software is published under the Open Software License. It is written in the PHP programming language with support for the MySQL database management system. PrestaShop is currently used by 300,000 shops worldwide and is available in 60 different languages.

Reviewer: ```suggestion
PrestaShop is a freemium, open source e-commerce solution written in PHP.
```
The rest sounds too markety.

## Features

* PHP 7.4
* MariaDB 10.4

> * Automatic TLS certificates

Reviewer: No composer-build?  Does it not use Composer?

Reviewer: Mention the Composer-based build, assuming that's still the case.  We do for other PHP apps.

## Customizations

> The following files and additions make the framework work.  If using this project as a reference for your own existing project, replicate the changes below to your project.

Reviewer: ```suggestion
The following changes have been made relative to Prestashop as it is downloaded from Prestashop.com. If using this project as a reference for your own existing project, replicate the changes below to your project.
```

(Or whatever the download origin is, or composer, or whatever.)

* [.platform/routes.yaml](.platform/routes.yaml): Platform.sh allows you to define the [routes](https://docs.platform.sh/configuration/routes.html).
* [.platform/services.yaml](.platform/services.yaml):  Platform.sh allows you to completely define and configure the topology and [services you want to use on your project](https://docs.platform.sh/configuration/services.html).
* [.platform.app.yaml](.platform.app.yaml): You control your application and the way it will be built and deployed on Platform.sh [via a single configuration file](https://docs.platform.sh/configuration/app-containers.html).

## References

* [Platform.sh post](https://platform.sh/blog/2019/java-hello-world-at-platform.sh/)
* [Prestashop](https://www.prestashop.com/en)
* [PHP at Platform.sh](https://docs.platform.sh/languages/php.html)

## How to install

1. Clone this repository
2. Create a new platform.sh project

```
platform project:create
```

3. Add specific environment variables

```
platform -p<project> variable:set ADMIN_EMAIL your@email.com

> platform -p<project> variable:set ADMIN_PASSWORD yourpassword

Reviewer: Wait, it's not installable without setting those first?

What we've done on other templates in that case is to bake in a default admin/password in the deploy hook or a .environment file or similar and then document "please change this thing!"  That way the one-click install button can still work, and we can eliminate this whole section.
